packages/orm/src/collections/ManyToOneReference.ts

```diff
@@ -136,11 +136,14 @@ export class ManyToOneReference<T extends Entity, U extends Entity, N extends ne
 
   // Internal method used by OneToManyCollection
   setImpl(other: U | N): void {
-    if (this.isLoaded && other === this.loaded) {
+    if (other?.isNewEntity ? other === this.loaded : this.id === other?.id) {
       return;
     }
 
-    const previousLoaded = this.loaded;
+    // we may not be loaded yet, but our previous entity might already be in the UoW
```

Comment:
Can you add a dedicated test for this to `ManyToOneReference.test.ts`? I could see this breaking / be worth testing explicitly while working on MTOR and not realizing it's only covered implicitly by other tests.